src/parse.y

```diff
@@ -877,17 +909,57 @@ void yyerror(const char* s) {
 
 	free(copy);
 	free_check_result(res);
+
+	// print line, replace tabs
+	unsigned short tabs_before_hinter = 0;
+	for (const char *c = current_line; *c != '\0'; ++c) {
+		if (*c == '\t') {
+			if (c - current_line < locp->first_column) {
+				tabs_before_hinter++;
+			}
+			printf("    ");
+			continue;
+		}
+
+		if (iscntrl((unsigned char)*c) && !isspace((unsigned char)*c)) {
+			printf("\n%sWarning%s: Line in question contains control character: 0x%x\n",
+			       color_warning(), color_reset(),
+			       *c);
+			return;
+		}
+
+		printf("%c", *c);
+	}
+
+	// print hinter
+	for (unsigned int i = 0; i < tabs_before_hinter; ++i) {
+		printf("    ");
+	}
+	for (unsigned int i = tabs_before_hinter + 1; i < locp->first_column; ++i) {
```

Comment:
My compiler is displaying a conversion warning for this problem: https://stackoverflow.com/questions/20236483/strange-gcc-short-int-conversion-warning

This (while obnoxious) seems to make the warning go away:

```
-       for (unsigned int i = tabs_before_hinter + 1; i < locp->first_column; ++i) {
+       for (unsigned short i = (unsigned short) (tabs_before_hinter +  1); i < locp->first_column; ++i) {
```